esphome/components/esp32/gpio.py

```
@@ -148,6 +153,20 @@ def validate_supports(value):
         )
 
     value = _esp32_validations[variant].usage_validation(value)
+
+    ignore_pin_validation_warning = value[CONF_IGNORE_PIN_VALIDATION_WARNING]
```

Similar to the `CONF_IGNORE_STRAPPING_WARNING`, this should also raise an error if you set `ignore_pin_validation_warning: true` and the pin is not one that would normally raise an error. This is to stop people putting it on every pin config.

---

Good idea! I've added this.